shell/completion.bash

```diff
@@ -175,16 +175,16 @@ __fzf_generic_path_completion() {
 
     [[ $base = *"/"* ]] && dir="$base"
     while true; do
-      if [[ -z "$dir" ]] || [[ -d "$dir" ]]; then
-        leftover=${base/#"$dir"}
+      if [[ -z "${dir-}" ]] || [[ -d "${dir-}" ]]; then
+        leftover=${base/#"${dir-}"}
         leftover=${leftover/#\/}
-        [[ -z "$dir" ]] && dir='.'
-        [[ "$dir" != "/" ]] && dir="${dir/%\//}"
-        matches=$(eval "$1 $(printf %q "$dir")" | FZF_DEFAULT_OPTS="--height ${FZF_TMUX_HEIGHT:-40%} --reverse --bind=ctrl-z:ignore $FZF_DEFAULT_OPTS $FZF_COMPLETION_OPTS $2" __fzf_comprun "$4" -q "$leftover" | while read -r item; do
+        [[ -z "${dir-}" ]] && dir='.'
```

Comment:
Do we need to write `${dir-}` after this line? `$dir` is guaranteed to be defined.

Reply:
bash complains about `dir` being unbound when using completion like `ls **`

Reply:
I mean **after this line**. I've tested and bash doesn't complain.

However, it's much simpler to just add `dir=` line before line 176 and just referencing it as `$dir` as before.

Reply:
```diff
diff --git a/shell/completion.bash b/shell/completion.bash
index ece458e..364c3b4 100644
--- a/shell/completion.bash
+++ b/shell/completion.bash
@@ -173,13 +173,14 @@ __fzf_generic_path_completion() {
     base=${cur:0:${#cur}-${#trigger}}
     eval "base=$base"
 
+    dir=
     [[ $base = *"/"* ]] && dir="$base"
     while true; do
-      if [[ -z "${dir-}" ]] || [[ -d "${dir-}" ]]; then
-        leftover=${base/#"${dir-}"}
+      if [[ -z "$dir" ]] || [[ -d "$dir" ]]; then
+        leftover=${base/#"$dir"}
         leftover=${leftover/#\/}
-        [[ -z "${dir-}" ]] && dir='.'
-        [[ "${dir-}" != "/" ]] && dir="${dir/%\//}"
+        [[ -z "$dir" ]] && dir='.'
+        [[ "$dir" != "/" ]] && dir="${dir/%\//}"
         matches=$(eval "$1 $(printf %q "${dir-}")" | FZF_DEFAULT_OPTS="--height ${FZF_TMUX_HEIGHT:-40%} --reverse --bind=ctrl-z:ignore ${FZF_DEFAULT_OPTS-} ${FZF_COMPLETION_OPTS-} $2" __fzf_comprun "$4" -q "$leftover" | while read -r item; do
           printf "%q " "${item%$3}$3"
         done)
@@ -193,8 +194,8 @@ __fzf_generic_path_completion() {
         printf '\e[5n'
         return 0
       fi
-      dir=$(dirname "${dir-}")
-      [[ "${dir-}" =~ /$ ]] || dir="${dir-}"/
+      dir=$(dirname "$dir")
+      [[ "$dir" =~ /$ ]] || dir="$dir"/
     done
   else
     shift
```

Reply:
ah! yes we this wont be needed if we instantiate an empty var.